pkg/doc/did/doc.go

```diff
@@ -166,9 +166,35 @@ type Service struct {
 	Properties      map[string]interface{}
 }
 
+// VerificationRelationship defines a verification relationship between DID subject and a verification method.
+type VerificationRelationship int
+
+const (
+	// VerificationRelationshipGeneral is a special case of verification relationship: when a public key
+	// defined in PublicKey is not used by any VerificationMethod.
```

Comment:
This comment isn't quite accurate. Any public key in the PublicKey section is a general key regardless of whether or not it is used by a VerificationMethod.

Reply:
I thought we want to distinguish the keys which are not referred from any Verification Method (e.g. assertion method). If not, it will simplify the code.

Reply:
We need the capability of knowing the full list of keys defined in the PublicKey section. (aka general).

Reply:
We can just read `Doc.PublicKey` field for that purpose. It's already present.

Reply:
We should have `general` mean the keys listed in PublicKey.

Reply:
Yes, all the keys listed in `PublicKey` are included in the return value of `Doc.VerificationMethods()` with `Relationship=VerificationRelationshipGeneral`.